src/test/java/net/sf/jabref/importer/fileformat/CopacImporterTestFiles.java

```diff
@@ -0,0 +1,76 @@
+package net.sf.jabref.importer.fileformat;
+
+import net.sf.jabref.*;
+import net.sf.jabref.bibtex.BibtexEntryAssert;
+import net.sf.jabref.importer.OutputPrinterToNull;
+import net.sf.jabref.model.entry.BibEntry;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RunWith(Parameterized.class)
+public class CopacImporterTestFiles {
+
+    private CopacImporter copacImporter;
+
+    @Parameter
+    public String fileName;
+
+
+    @Before
+    public void setUp() {
+        Globals.prefs = JabRefPreferences.getInstance();
+        copacImporter = new CopacImporter();
+    }
+
+    @Parameters(name = "{0}")
+    public static Collection<String> fileNames() {
+        List<String> files = new ArrayList<>();
+        File d = new File(System.getProperty("user.dir") + "/src/test/resources/net/sf/jabref/importer/fileformat");
+        for (File f : d.listFiles()) {
+            files.add(f.getName());
+        }
+        return files.stream().filter(n -> n.startsWith("CopacImporterTest")).collect(Collectors.toList());
+    }
+
+    @Test
+    public void testIsRecognizedFormat() throws IOException {
+        try (InputStream stream = CopacImporterTest.class.getResourceAsStream(fileName)) {
+            Assert.assertTrue(copacImporter.isRecognizedFormat(stream));
+        }
+    }
+
+    @Test
+    public void testImportEntries() throws IOException {
+        try (InputStream copacStream = CopacImporterTest.class.getResourceAsStream(fileName)) {
+
+            List<BibEntry> copacEntries = copacImporter.importEntries(copacStream, new OutputPrinterToNull());
+            fileName = fileName.replace(".txt", "");
+
+            int size = copacEntries.size();
+
+            if (size != 1) {
+                for (int i = 1; i <= size; i++) {
+                    BibtexEntryAssert.assertEquals(CopacImporterTest.class, "Bib" + fileName + "-" + i + ".bib",
```

Comment:
Would it be possible to name the copac files `CopacImporter....txt` and the bib files `CopacImporter....bib`. The prefix `Bib` is not necessary.

Please remove the `Bib` prefix and adapt the filter on line 46: Add `.filter(n -> n.endsWith(".txt)` before `.collect`.

Adapt line 61 to replace `.txt` by `.bib` and just pass `fileName` here.

In case there are multiple entries in your list, possibly BibTexEntryAssert has to be modified. We have to discuss then.